rocks-db/src/clients/dump_client.rs

```diff
@@ -337,7 +338,7 @@ impl Dumper for Storage {
                     synchronizer_metrics.inc_num_of_records_written("authority", 1);
                 }
             }
-            if !rx.is_empty() {
+            if cancellation_token.is_cancelled() {
```

Comment:
the cancellation is barely reachable inside this huge function

Reply:
not really, it will cancel upon subsequent iterator accesses, which are pretty fast